ghmirror/core/mirror_requests.py

```diff
@@ -118,49 +181,40 @@ def online_request(method, url, auth, data=None, url_params=None):
         if last_mod is not None:
             headers['If-Modified-Since'] = last_mod
 
-    resp = requests.request(method=method,
-                            url=url,
-                            headers=headers,
-                            timeout=REQUESTS_TIMEOUT,
-                            params=parameters)
+    resp = _online_request(method=method,
+                           url=url,
+                           headers=headers,
+                           parameters=parameters,
+                           cached_response=cached_response)
 
     if resp.status_code == 304:
-        if len(cached_response.json()) == per_page_elements and \
-           not cached_response.links:
+        return _handle_not_changed(cached_response, per_page_elements,
+                                   headers, method, url, parameters,
+                                   cache, cache_key)
+
+    _cache_response(resp, cache, cache_key)
```

Comment:
If I understand correctly the code, you still don't know if you have a cache miss as `_online_request` you are going to see if result code is 304, isn't it?

Reply:
I'm adding the headers to the response that comes from the API, but if the response code is 304, we will be returning either a new request made to the api with the correct headers (Cache miss) or the response that came from the Cache with the CACHE_HIT header, and the original response is discarded. I'm trying to look for a way to clarify the code

Reply:
Hmm, yes, headers will be rewritten in the `304` `if` branch but you would still get an incorrect log message. That has to be the reason why tests are working.

Reply:
Didn't saw that, 100% true, this fix should log the correct status of the request

Reply:
```
    _cache_response(resp, cache, cache_key)
```
That should go inside the conditional as we will be serving cached data otherwise

Please add a nice comment explaining how we have arrived into that conditional since the logic is inside `_online_request`